### PR TITLE
Deleted message event

### PR DESCRIPTION
**Pull request by mannguyen0107**
#### Bug fixes 
- Add undefined type to `message` param of `getMessageId`
- Fix a typo in `EditedMessage` event docs

#### Features:
- Deleted message event

### CODE REVIEW

File: gramjs/Utils.ts [hunk 1]
```diff
@@ -1172,7 +1172,7 @@ export function  _getEntityPair(entityId, entities, cache, getInputPeer = getInp
 */
 
 export function getMessageId(
-    message: number | Api.TypeMessage | MessageIDLike
+    message: number | Api.TypeMessage | MessageIDLike | undefined
 ): number | undefined {
     if (message === null || message === undefined) {
```
Comment (painor) [inline gramjs/Utils.ts:1177]: you can just do if (!message) and it'll catch all of those

File: gramjs/events/DeletedMessage.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * was deleted.
+ *
+ * @remarks
+ * Telegram **does not** send information about *where* a message
+ * was deleted if it occurs in private conversations with other users
+ * or in small group chats, because message IDs are *unique* and you
+ * can identify the chat with the message ID alone if you saved it
+ * previously.
+ *
+ * Telethon **does not** save information of where messages occur,
```
Comment (painor) [inline gramjs/events/DeletedMessage.ts:17]: telethon <.<

File: gramjs/events/DeletedMessage.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+ * so it cannot know in which chat a message was deleted (this will
+ * only work in channels, where the channel ID *is* present).
+ *
+ * This means that the `chats=` parameter will not work reliably,
```
Comment (painor) [inline gramjs/events/DeletedMessage.ts:21]: chats: not chats=

File: gramjs/events/DeletedMessage.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+export class DeletedMessageEvent extends EventCommon {
+    deletedIds: number[];
+    peer?: EntityLike;
+
+    constructor(deletedIds: number[], peer?: EntityLike) {
+        super({
+            chatPeer: peer,
+            msgId: (deletedIds || [0])[0],
```
Comment (painor) [inline gramjs/events/DeletedMessage.ts:66]: could probably use ternary operation here instead of this

**Review by painor — CHANGES_REQUESTED** [review]

File: gramjs/events/DeletedMessage.ts (hunk 2 above)
Comment (mannguyen0107) [reply gramjs/events/DeletedMessage.ts:17]: oops copy paste is bad haha

File: gramjs/events/DeletedMessage.ts (hunk 4 above)
Comment (mannguyen0107) [reply gramjs/events/DeletedMessage.ts:66]: is there a performance difference?

Comment (painor) [reply gramjs/events/DeletedMessage.ts]: maybe, but ternary also looks nicer than [0][0]

Comment (mannguyen0107) [reply gramjs/events/DeletedMessage.ts]: aight pushed all the changes you commented

**Review by painor — APPROVED** [review]

**Pull request merged**